service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/create/CreateAzureVmStep.java

```diff
@@ -0,0 +1,173 @@
+import com.azure.resourcemanager.network.models.Network;
+import com.azure.resourcemanager.network.models.PublicIpAddress;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateAzureVmStep implements Step {
+
+  private static final Logger logger = LoggerFactory.getLogger(CreateAzureIpStep.class);
+  private final AzureConfiguration azureConfig;
+  private final CrlService crlService;
+  private final ControlledAzureVmResource resource;
+  private final AzureCloudContext azureCloudContext;
+  private final ResourceDao resourceDao;
+
+  public CreateAzureVmStep(
+      AzureConfiguration azureConfig,
+      AzureCloudContext azureCloudContext,
+      CrlService crlService,
+      ControlledAzureVmResource resource,
+      ResourceDao resourceDao) {
+    this.azureConfig = azureConfig;
+    this.azureCloudContext = azureCloudContext;
+    this.crlService = crlService;
+    this.resource = resource;
+    this.resourceDao = resourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
+
+    final ControlledAzureIpResource ipResource =
+        resourceDao
+            .getResource(resource.getWorkspaceId(), resource.getIpId())
+            .castToControlledResource()
+            .castToAzureIpResource();
+
+    final ControlledAzureDiskResource diskResource =
+        resourceDao
+            .getResource(resource.getWorkspaceId(), resource.getDiskId())
+            .castToControlledResource()
+            .castToAzureDiskResource();
+
+    final ControlledAzureNetworkResource networkResource =
+        resourceDao
+            .getResource(resource.getWorkspaceId(), resource.getNetworkId())
+            .castToControlledResource()
+            .castToAzureNetworkResource();
+
+    try {
+      Disk existingAzureDisk =
+          computeManager
+              .disks()
+              .getByResourceGroup(
+                  azureCloudContext.getAzureResourceGroupId(), diskResource.getDiskName());
+
+      PublicIpAddress existingAzureIp =
+          computeManager
+              .networkManager()
+              .publicIpAddresses()
+              .getByResourceGroup(
+                  azureCloudContext.getAzureResourceGroupId(), ipResource.getIpName());
+
+      Network existingNetwork =
+          computeManager
+              .networkManager()
+              .networks()
+              .getByResourceGroup(
+                  azureCloudContext.getAzureResourceGroupId(), networkResource.getNetworkName());
+
+      computeManager
+          .virtualMachines()
+          .define(resource.getVmName())
+          .withRegion(resource.getRegion())
+          .withExistingResourceGroup(azureCloudContext.getAzureResourceGroupId())
+          .withExistingPrimaryNetwork(existingNetwork)
```

Comment:
Not for this PR - just noting there's an IA ticket to make a `NetworkInterface` resource in CRL/WSM. Right now, creating a VM implicitly creates a network interface which is untracked by WSM, causing issues at deletion time.

https://broadworkbench.atlassian.net/browse/IA-3097